src/Interpreters/ActionsDAG.cpp

```diff
@@ -2511,11 +2511,21 @@ FindOriginalNodeForOutputName::FindOriginalNodeForOutputName(const ActionsDAGPtr
         /// find input node which refers to the output node
         /// consider only aliases on the path
         const auto * node = output_node;
-        while (node && node->type == ActionsDAG::ActionType::ALIAS)
+        while (node)
         {
-            /// alias has only one child
-            chassert(node->children.size() == 1);
-            node = node->children.front();
+            if (node->type == ActionsDAG::ActionType::ALIAS)
+            {
+                node = node->children.front();
+            }
+            /// materiailze() function can occur when dealing with views
+            /// TODO: not sure if it should be done here, looks too generic place
+            else if (node->type == ActionsDAG::ActionType::FUNCTION && node->function_base->getName() == "materialize")
+            {
+                chassert(node->children.size() == 1);
```

Comment:
If a user writes `materialize(a)(b)`, will it be checked earlier?

Reply:
Sorry, don't get your question. But added explicitly query into the test with `materialize()` to check explicitly that usage of `materialize()` doesn't affect the optimization.

> If a user writes materialize(a)(b), will it be checked earlier?
(1) how `materialize()` can have 2 parameters?
(2) what do you mean by check earlier? Here we check it while walking through `ActionsDAG` for all expressions which are between preliminary `DistinctStep` (redundant) and final `DistinctStep`(first distinct from data source)